proxy/envoy/discovery.go

```diff
@@ -396,41 +396,44 @@ func (ds *DiscoveryService) ListEndpoints(request *restful.Request, response *re
 		}
 		var err error
 		if out, err = json.MarshalIndent(hosts{Hosts: hostArray}, " ", " "); err != nil {
-			errorResponse(response, http.StatusInternalServerError, err.Error())
+			errorResponse(response, http.StatusInternalServerError, "SDS " + err.Error())
 			return
 		}
 		ds.sdsCache.updateCachedDiscoveryResponse(key, out)
 	}
 	writeResponse(response, out)
 }
 
-func (ds *DiscoveryService) parseRole(request *restful.Request) (proxy.Node, error) {
-	if sc := request.PathParameter(ServiceCluster); sc != ds.Mesh.IstioServiceCluster {
-		return proxy.Node{}, fmt.Errorf("unexpected %s %q", ServiceCluster, sc)
+func (ds *DiscoveryService) parseDiscoveryRequest(request *restful.Request) (string, string, proxy.Node, error) {
+	cluster := request.PathParameter(ServiceCluster)
+	if cluster != ds.Mesh.IstioServiceCluster {
+		return cluster, "", proxy.Node{}, fmt.Errorf("unexpected %s %q", ServiceCluster, cluster)
 	}
 
 	node := request.PathParameter(ServiceNode)
 	role, err := proxy.ParseServiceNode(node)
 	if err != nil {
-		return role, multierror.Prefix(err, fmt.Sprintf("unexpected %s: ", ServiceNode))
+		return cluster, node, role, multierror.Prefix(err, fmt.Sprintf("unexpected %s: ", ServiceNode))
 	}
-	return role, nil
+	return cluster, node, role, nil
 }
 
 // ListClusters responds to CDS requests for all outbound clusters
 func (ds *DiscoveryService) ListClusters(request *restful.Request, response *restful.Response) {
 	key := request.Request.URL.String()
 	out, cached := ds.cdsCache.cachedDiscoveryResponse(key)
 	if !cached {
-		role, err := ds.parseRole(request)
+		cluster, node, role, err := ds.parseDiscoveryRequest(request)
 		if err != nil {
-			errorResponse(response, http.StatusNotFound, err.Error())
+			errorResponse(response, http.StatusNotFound, "CDS " + err.Error())
 			return
 		}
+		glog.V(3).Infof("CDS Discovery request to ListClusters for service_cluster %s, service_node %s, role %s",
+			cluster, node, role.Type)
```

Comment:
Remove? Even if accidentally turned on, this will flood the system with logs, at a 100ms refresh frequency